packages/valory/skills/abstract_round_abci/background_round.py

```diff
@@ -0,0 +1,512 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2022 Valory AG
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""This module contains the background behaviour and round classes."""
+from enum import Enum
+from mailbox import Message
+from typing import Any, Callable, Dict, Generator, Optional, Tuple, cast, List
+
+from hexbytes import HexBytes
+
+from packages.valory.contracts.gnosis_safe.contract import GnosisSafeContract, SafeOperation
+from packages.valory.contracts.multisend.contract import MultiSendContract, MultiSendOperation
+from packages.valory.contracts.service_registry.contract import ServiceRegistryContract
+from packages.valory.protocols.contract_api import ContractApiMessage
+from packages.valory.skills.abstract_round_abci.base import (
+    BaseSynchronizedData,
+    BaseTxPayload,
+    CollectSameUntilThresholdRound,
+)
+from packages.valory.skills.abstract_round_abci.behaviour_utils import (
+    AsyncBehaviour,
+    BaseBehaviour,
+)
+
+# setting the safe gas to 0 means that all available gas will be used
+# which is what we want in most cases
+# more info here: https://safe-docs.dev.gnosisdev.com/safe/docs/contracts_tx_execution/
+_SAFE_GAS = 0
+
+# hardcoded to 0 because we don't need to send any ETH when handing over multisig ownership
+_ETHER_VALUE = 0
+
+NULL_ADDRESS: str = "0x" + "0" * 40
+MAX_UINT256 = 2 ** 256 - 1
+
+# copied from transaction_settlement_abci/payload_tools.py to avoid cyclic dependency
+def hash_payload_to_hex(  # pylint: disable=too-many-arguments, too-many-locals
```

Comment:
Hm so many dependencies... we could extract this whole background setup as a `background_abci` into its own skill and introduce it via the config level - as suggested below for `transaction_settlement_abci` too. That would allow us to define arbitrary skills (with one round and one behaviour) as background skills.